### PR TITLE
MongoDB native driver connection

### PR DESCRIPTION
**Pull request by kabachook**
- Native mongo connection
- New event format

### CODE REVIEW

**Comment by kabachook** [comment]
This pull request **introduces 1 alert** when merging 215306e9e6e60f6aa06d660d432640b22f89a1ab into ba0b9ba46e8abfdf1659084a4befedfbcca78ab4 - [view on LGTM.com](https://lgtm.com/projects/g/codex-team/hawk.workers/rev/pr-10cc7b0ec03e1293e1916935b9a9fa2258485c72)

**new alerts:**

* 1 for Unused variable, import, function or class

---

This project has automated code review enabled, but doesn't use the <a href="https://github.com/apps/lgtm-com">LGTM GitHub App</a>. Migrate over by <a href="https://github.com/apps/lgtm-com/installations/new">installing the app</a>. Read about the <a href="https://lgtm.com/blog/announcing_github_apps_migration">benefits of migrating to GitHub Apps</a> in the blog.

---

*Comment posted by [LGTM.com](https://lgtm.com)*

File: lib/db/controller.js [hunk 1]
```diff
@@ -0,0 +1,76 @@
...
+   *
+   * @param {string|ObjectID} projectId - project id
+   * @param {eventSchema} eventData - event data
+   * @returns {Promise<mongodb.ObjectID>} saved event id
+   * @throws {ValidationError} if `projectID` is not provided or invalid
+   * @throws {ValidationError} if `eventData` is not a valid object
+   */
+  async saveEvent(projectId, eventData) {
+    if (!projectId || !mongodb.ObjectID.isValid(projectId)) {
+      throw new ValidationError('projectID is not set ot valid');
```
Comment (neSpecc) [inline lib/db/controller.js]:
```suggestion
      throw new ValidationError('Project ID is invalid or missed');
```

File: lib/db/models/event.js [hunk 2]
```diff
@@ -1,127 +1,50 @@
...
+
+let eventSchema = yup.object().shape({
+  catcherType: yup.string().required(),
+  payload: yup
+    .object()
+    .shape({
+      title: yup.string().required(),
+      timestamp: yup.date().required(),
+      level: yup.number().integer(),
+      // .required(), // enable when fix senders
```
Comment (neSpecc) [inline lib/db/models/event.js]: лучше через todo + issue

File: lib/db/models/event.js [hunk 3]
```diff
@@ -1,127 +1,50 @@
...
-
-    /**
-     * @optional
-     * Any additional data
-     */
-    context: Schema.Types.Mixed
-  }
+const yup = require('yup');
+
+let eventSchema = yup.object().shape({
```
Comment (neSpecc) [inline lib/db/models/event.js]: пропали все доки. Надо сохранить описание полей

File: package.json [hunk 4]
```diff
@@ -17,8 +17,11 @@
         "amqplib": "^0.5.3",
         "debug": "^4.1.1",
         "dotenv": "^6.2.0",
+        "jsonwebtoken": "^8.5.1",
+        "mongodb": "^3.2.7",
         "mongoose": "^5.4.4",
```
Comment (neSpecc) [inline package.json]: mongoose выпилен?

File: tools/generateToken.js [hunk 5]
```diff
@@ -0,0 +1,21 @@
+#!/usr/bin/env node
+
+/**
+ * Generate token tool
```
Comment (neSpecc) [inline tools/generateToken.js]: описание не достаточно. Напиши подробнее и понятнее, для чего используется и как использовать

**Comment by kabachook** [comment]
This pull request **introduces 1 alert** when merging 918ed7ed7fbea68cc65a64c0ee358d618d6beda8 into ba0b9ba46e8abfdf1659084a4befedfbcca78ab4 - [view on LGTM.com](https://lgtm.com/projects/g/codex-team/hawk.workers/rev/pr-20781eb04f8466f6c18aa4cd8220ae5a44f4f934)

**new alerts:**

* 1 for Unused variable, import, function or class

---

This project has automated code review enabled, but doesn't use the <a href="https://github.com/apps/lgtm-com">LGTM GitHub App</a>. Migrate over by <a href="https://github.com/apps/lgtm-com/installations/new">installing the app</a>. Read about the <a href="https://lgtm.com/blog/announcing_github_apps_migration">benefits of migrating to GitHub Apps</a> in the blog.

---

*Comment posted by [LGTM.com](https://lgtm.com)*

File: lib/db/controller.js [hunk 6]
```diff
@@ -0,0 +1,76 @@
...
+
+/**
+ * Database connection singleton
+ *
+ * @param {mongodb.MongoClient} connection - MongoDB connection
+ * @param {mongodb.Db} db - MongoDB connection database instance
+ */
+class Controller {
+  /**
+   * Connect to databse
```
Comment (nikmel2803) [inline lib/db/controller.js]:
```suggestion
   * Connect to database
```

File: lib/db/controller.js [hunk 7]
```diff
@@ -0,0 +1,76 @@
...
+   *
+   * @returns {Promise<void>}
+   */
+  async close() {
+    this.db = null;
+    return this.connection.close();
+  }
+
+  /**
+   * Save event to databse
```
Comment (nikmel2803) [inline lib/db/controller.js]:
```suggestion
   * Save event to database
```

File: tools/generateToken.js [hunk 8]
```diff
@@ -0,0 +1,25 @@
+#!/usr/bin/env node
+/* eslint-disable require-jsdoc*/
+
+/**
+ * Generate token tool
+ * Used to generate valid token for catcher for specific `projectId`
+ * Convinient for testing purposes, e.g. setting token for bombers
```
Comment (nikmel2803) [inline tools/generateToken.js]:
```suggestion
 * Convenient for testing purposes, e.g. setting token for bombers
```

**Pull request merged**